http/middleware.go

```diff
@@ -39,8 +39,20 @@ func WrapHandler(waf coraza.WAF, l Logger, h http.Handler) http.Handler {
 			return
 		}
 
+		i := &rwInterceptor{w: w, tx: tx, StatusCode: 200}
 		// We continue with the other middlewares by catching the response
-		h.ServeHTTP(wrap(w, r, tx), r)
+		h.ServeHTTP(i.wrap(w, tx), r)
+
+		for k, vv := range w.Header() {
+			for _, v := range vv {
+				tx.AddResponseHeader(k, v)
+			}
+		}
+
+		if it := tx.ProcessResponseHeaders(i.StatusCode, r.Proto); it != nil {
```

Comment:
So as far as I understand from the code, previously we check for a response headers interruption before sending the status code (I guess to send 403). Now we don't, we send the status code (i.e. WriteHeader is called) and then check for an interruption. This will call a subsequent `Writeheader` with 503 on deny and I am not sure that works.

Reply:
Of course you can always prove me wrong with a unit test :p.